site/docs/expressions/scalar_functions.md

```diff
@@ -1,52 +1,118 @@
-| Description        | Additional description of function for implementers or users. Should be written human readable to allow exposure to end users. Presented as a map with language => description mappings. E.g. `{ "en": "This adds two numbers together.", "fr": "cela ajoute deux nombres"}`. | Optional                            |
-| Output Type        | The output type is expected to be a physical type of the expression. There are two categories of output type:<br />Direct: Output type is declared<br />Complex: Behavior definitions are either declared within the specification or embedded in the signature using [TBD]. | Required                            |
-| Implementation Map | A map of implementation locations for one or more implementations of the given function. Each key is a function implementation type. Implementation types include examples such as: AthenaArrowLambda, TrinoV361Jar, ArrowCppKernelEnum, GandivaEnum, LinkedIn Transport Jar, etc. [Definition TBD]. Implementation type has one or more properties associated with retrieval of that implementation. | Optional                            |
 
 
+## Argument Types
 
-## Argument Properties
+There are two main types of arguments: Value Arguments and Type Arguments. 
 
-| Property      | Description                                                  |
-| ------------- | ------------------------------------------------------------ |
-| Physical Type | The physical types this argument requires. Physical types are used here so that different binds are used when working in a physical plan depending on the execution engines capabilities. In common cases, the system default representation is used. |
-| Constant      | Whether this argument is required to be a constant for invocation. For example, in some system a regular expression pattern would only be accepted as a literal and not a column value reference. |
+* Value Arguments: Arguments that where the function will refer to data value, could be a constant (literal-based expression) or variable (reference-based expression). This is the most common type of argument. Value arguments are not available in output derivation. Value arguments can be declared in one of two ways: concrete or parameterized. Concrete Types are either simple types or compound types with all parameters fully defined (without any parameters). Examples include i32, fp32, VARCHAR(20), List&lt;fp32&gt;, etc. Parameterized types are discussed further below.
+* Type Arguments: Type arguments are used to inform the evaluation and/or type derivation of the function. For example, you might have a function which is `truncate(<type> DECIMAL(P0,S0), <value> DECIMAL(P1, S1), <value> i32)`. This function declares two value arguments (arg1 and arg2) and type argument. The type argument's value can be used in output type determination. (The types of the values arguments can also be used for output type derivation.)
```

Comment:
```suggestion
* Value Arguments: An argument that refers to a data value.  This could be a constant (a literal expression defined in the plan) or variable (A reference expression that references data being processed by the plan). This is the most common type of argument. Value arguments are not available in output derivation. Value arguments can be declared in one of two ways: concrete or parameterized. Concrete Types are either simple types or compound types with all parameters fully defined (without referencing any type arguments). Examples include i32, fp32, VARCHAR(20), List&lt;fp32&gt;, etc. Parameterized types are discussed further below.
* Type Arguments: Type arguments are used to inform the evaluation and/or type derivation of the function. For example, you might have a function which is `truncate(<type> DECIMAL(P0,S0), <value> DECIMAL(P1, S1), <value> i32)`. This function declares two value arguments and a type argument. The type argument's value can be used to determine the output type (The types of the values arguments can also be used for output type derivation.)
```

I agree with what you are saying, I am just (hopefully) clearing up the wording.  `literal-based expression` and `reference-based expression` confused me for a while but I get what you are saying.  I removed `(arg1 and arg2)` because it isn't clear to me what you were trying to say.

Reply:
nice improvements, thanks